alg/teca_temporal_reduction.h

```diff
@@ -0,0 +1,239 @@
+ *
+ *     time increments: daily, monthly, seasonal, yearly, n_steps
+ *     reduction operators: average, summation, minimum, maximum
+ *
+ * The output time axis will be defined using the selected increment.
+ * The output data will be accumulated/reduced using the selected
+ * operation.
+ *
+ * The set_use_fill_value method controls how invalid or missing values are
+ * treated. When set to 1, NetCDF CF fill values are detected and handled.
+ * This is the default. If it is known that the dataset has no invalid or
+ * missing values one may set this to 0 for faster processing. By default the
+ * fill value will be obtained from metadata stored in the NetCDF CF file
+ * (_FillValue). One may override this by explicitly calling set_fill_value
+ * method with the desired fill value.
+ *
+ * For minimum and maximum operations, at given grid point only valid values
+ * over the interval are used in the calculation. If there are no valid
+ * values over the interval at the grid point it is set to the fill_value.
+ *
+ * For the averaging operation, during summation missing values are treated
+ * as 0.0 and a per-grid point count of valid values over the interval is
+ * maintained and used in the average. Grid points with no valid values over
+ * the interval are set to the fill value.
+ */
+class TECA_EXPORT teca_temporal_reduction : public teca_threaded_algorithm
+{
+public:
+    TECA_ALGORITHM_STATIC_NEW(teca_temporal_reduction)
+    TECA_ALGORITHM_DELETE_COPY_ASSIGN(teca_temporal_reduction)
+    TECA_ALGORITHM_CLASS_NAME(teca_temporal_reduction)
+    ~teca_temporal_reduction();
+
+    // report/initialize to/from Boost program options
+    // objects.
+    TECA_GET_ALGORITHM_PROPERTIES_DESCRIPTION()
+    TECA_SET_ALGORITHM_PROPERTIES()
+
+    /** @name point_arrays
+     * Set the list of arrays to reduce
+     */
+    ///@{
+    TECA_ALGORITHM_VECTOR_PROPERTY(std::string, point_array)
+
+    /** @name operator_name
+     * Set the reduction operator
+     * default "None"
+     * It can be:
+     *           average
+     *           summation
+     *           minimum
+     *           maximum
+     */
+    ///@{
+    int set_operator(const std::string &op);
+    ///@}
+
+    /** @name interval_name
+     * Set the type of interval iterator to create
+     * default "None"
+     * It can be:
+     *           daily
+     *           monthly
+     *           seasonal
+     *           yearly
+     *           n_steps
+     * For the n_steps iterator replace n with the
+     * desired number of steps (e.g. 8_steps)
+     */
+    ///@{
+    int set_interval(const std::string &interval);
+    ///@}
+
+    /** @name fill_value
+     * Set the _fillValue attribute for the output data
+     * default -1
+     */
+    ///@{
+    TECA_ALGORITHM_PROPERTY(double, fill_value)
+
+    /** @name use_fill_value
+     * Control how invalid or missing values are treated
+     * default 1
+     */
+    ///@{
+    TECA_ALGORITHM_PROPERTY(int, use_fill_value)
```

Comment:
These can be removed. The missing value handling should all be automatic when used with the teca_valid_value mask. The Python temporal reduction predates the teca_valid_value_mask and these are no longer needed.

Reply:
@burlen Should I remove only line 208 or lines 201 and 208?